### PR TITLE
Unify ugettext imports and aliases

### LINKED ISSUE #1574: Unify ugettext, ugettext_lazy and variants usage
Unify `ugettext`, `ugettext_lazy` and variants usage. It's not always used correctly (e.g non-lazy ugettext on module-level) and sometimes `_` references to the lazy variant and vice versa. We need a clear guideline.

I'd propose:
- `ugettext_lazy` -> `_` (as this is used most often, e.g in models, forms and constants)
- explicit `ugettext` (and others) calls to make sure one sees the differences

Opinions?

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #1574 

This then allows us to have a convention which we can enforce through linting eventually. I also found ~4-6 more wrong usages where `ugettext` got used on import-level and fixed a few cases where string-formatting was applied to the unwrapped string where I found them.

### CODE REVIEW

File: src/olympia/addons/widgets.py [hunk 1]
```diff
@@ -45,8 +45,9 @@ def render(self, name, value, attrs=None):
         miscs = Category.objects.filter(misc=True).values_list('id', flat=True)
         for c in self.choices:
             if c[0] in miscs:
-                other = (c[0],
-                         _("My add-on doesn't fit into any of the categories"))
+                msg = ugettext(
+                    "My add-on doesn't fit into any of the categories")
```
Comment (eviljeff) [inline src/olympia/addons/widgets.py]: `"` - double quotes

File: src/olympia/devhub/forms.py [hunk 2]
```diff
@@ -121,10 +122,10 @@ class LicenseForm(AMOModelForm):
         choices=[], coerce=int,
         widget=LicenseRadioSelect(attrs={'class': 'license'}))
     name = forms.CharField(widget=TranslationTextInput(),
-                           label=_lazy(u"What is your license's name?"),
-                           required=False, initial=_lazy('Custom License'))
+                           label=_(u"What is your license's name?"),
```
Comment (eviljeff) [inline src/olympia/devhub/forms.py]: `"` double quotes

Comment (eviljeff) [inline src/olympia/devhub/forms.py]: and more in this file

File: src/olympia/activity/serializers.py [hunk 3]
```diff
@@ -29,7 +29,8 @@ def get_comments(self, obj):
 
     def get_action_label(self, obj):
         log = obj.log()
-        return _(u'Review note') if not hasattr(log, 'short') else log.short
+        default = ugettext(u'Review note')
```
Comment (EnTeQuAk) [inline src/olympia/activity/serializers.py]: I doubt it was intendedly a `ugettext_lazy` before but by accident. All other instances of `get_action_label`s result are being used as a regular string so laziness didn't do anything here.

File: src/olympia/addons/widgets.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/addons/widgets.py]: They're all over the place. To be honest, I started but eventually gave up changing them too all the time.

**Comment by EnTeQuAk** [comment]
I'm back! This now has quite a few conflicts (nothing major though), @eviljeff did you have time to go through the patch completely? I'll cleanup the conflicts and I think now, with fresh energy I'll just change all the touched `"` to `'"` to get this stone rolling once and for all too :)

File: src/olympia/legacy_api/views.py [hunk 4]
```diff
@@ -36,7 +36,7 @@
 
 
 ERROR = 'error'
-OUT_OF_DATE = ugettext_lazy(
+OUT_OF_DATE = _(
```
Comment (eviljeff) [inline src/olympia/legacy_api/views.py]: should be `ugettext` I guess as it's an API response.

Comment (EnTeQuAk) [inline src/olympia/legacy_api/views.py]: I disagree, it's being translated on run-time when we call `OUT_OF_DATE.format(…)` so being lazy here is perfectly fine (and correct).

**Comment by EnTeQuAk** [comment]
Fixed conflicts, rebased and fixed all review comments (and lots more `"` vs `'` replacements). I'm going to merge this if travis agrees.

**Pull request merged**